File: src/main/java/com/mo/mediaodyssey/auth/config/AuthRoutes.java
package com.mo.mediaodyssey.auth.config;

/**
 * Shared auth routes and query parameter names.
 *
 * Keep route strings here so the auth controller, security config, rate
 * limiter, and auth pages all point at the same paths.
 */
public final class AuthRoutes {

    public static final class Api {
        public static final String AUTH_API_PREFIX = "/api/auth";
        public static final String LOGIN = AUTH_API_PREFIX + "/login";
        public static final String REGISTER = AUTH_API_PREFIX + "/register";
        public static final String VERIFY = AUTH_API_PREFIX + "/verify";
        public static final String RESEND = AUTH_API_PREFIX + "/resend";
        public static final String PASSWORD_FORGOT = AUTH_API_PREFIX + "/password/forgot";
        public static final String PASSWORD_RESET = AUTH_API_PREFIX + "/password/reset";
        public static final String LOGOUT = AUTH_API_PREFIX + "/logout";

    }

    public static final class Page {
        public static final String AUTH_PAGE_ROOT = "/auth";
        public static final String LOGIN = AUTH_PAGE_ROOT + "/login";
        public static final String SIGNUP = AUTH_PAGE_ROOT + "/signup";
        public static final String FORGOT = AUTH_PAGE_ROOT + "/forgot";
        public static final String RESET = AUTH_PAGE_ROOT + "/reset";
        public static final String VERIFY = AUTH_PAGE_ROOT + "/verify";
        public static final String RESEND = AUTH_PAGE_ROOT + "/resend";
        public static final String LOGOUT = AUTH_PAGE_ROOT + "/logout";
        public static final String OAUTH2_AUTHORIZATION = AUTH_PAGE_ROOT + "/oauth2/authorization";
        public static final String OAUTH2_CALLBACK = AUTH_PAGE_ROOT + "/oauth2/callback/*";

    }

    public static final class QueryParams {
        public static final String SESSION_EXPIRED = "sessionExpired";
        public static final String OAUTH_ERROR = "oauthError";

    }
}

File: src/main/resources/static/auth/auth-routes.js
window.AuthRoutes = (() => {
    const AUTH_API_PREFIX = "/api/auth";
    const AUTH_PAGE_ROOT = "/auth";

    return Object.freeze({
        Api: Object.freeze({
            AUTH_API_PREFIX,
            LOGIN: AUTH_API_PREFIX + "/login",
            REGISTER: AUTH_API_PREFIX + "/register",
            VERIFY: AUTH_API_PREFIX + "/verify",
            RESEND: AUTH_API_PREFIX + "/resend",
            PASSWORD_FORGOT: AUTH_API_PREFIX + "/password/forgot",
            PASSWORD_RESET: AUTH_API_PREFIX + "/password/reset",
            LOGOUT: AUTH_API_PREFIX + "/logout",
        }),
        Page: Object.freeze({
            AUTH_PAGE_ROOT,
            LOGIN: AUTH_PAGE_ROOT + "/login",
            SIGNUP: AUTH_PAGE_ROOT + "/signup",
            FORGOT: AUTH_PAGE_ROOT + "/forgot",
            RESET: AUTH_PAGE_ROOT + "/reset",
            VERIFY: AUTH_PAGE_ROOT + "/verify",
            RESEND: AUTH_PAGE_ROOT + "/resend",
            LOGOUT: AUTH_PAGE_ROOT + "/logout",
            OAUTH2_AUTHORIZATION: AUTH_PAGE_ROOT + "/oauth2/authorization",
            OAUTH2_CALLBACK: AUTH_PAGE_ROOT + "/oauth2/callback/*",
        }),
        QueryParams: Object.freeze({
            SESSION_EXPIRED: "sessionExpired",
            OAUTH_ERROR: "oauthError",
        }),
    });
})();
